### PR TITLE
fix(FEC-9508): Playback doesn't work on IE11 windows 7 with plugins

### PR DESCRIPTION
**Pull request by Yuvalke**
**issue**: missing methods from interface and proxy issue
**solution**: Add the methods from engine interface to be compatible with all the other engines.
init the members on init for proxy issue which all the members should exist(initialised) before proxy created.

Solve FEC-9508

### Description of the Changes

Please add a detailed description of the change, weather it's an enhancement or a bugfix.
If the PR is related to an open issue please link to it.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/flash.js [hunk 1]
```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+  /**
+   * Gets the loop value of the video element.
+   * @returns {boolean} - The loop value.
+   * @public
+   */
+  get loop(): boolean {
+    return false;
+  }
+
+  isPictureInPictureSupported(): void {}
```
Comment (OrenMe) [inline src/flash.js]: Shouldn’t this return Boolean?

File: src/flash.js [hunk 2]
```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+   * 2 = HAVE_CURRENT_DATA - data for the current playback position is available, but not enough data to play next frame/millisecond.
+   * 3 = HAVE_FUTURE_DATA - data for the current and at least the next frame is available.
+   * 4 = HAVE_ENOUGH_DATA - enough data available to start playing.
+   */
+  get readyState(): number {
+    if (!this._api) {
+      return 0;
+    }
+    return NaN;
+  }
```
Comment (OrenMe) [inline src/flash.js]: This api can’t return NAN, and in general for flash either return 0 for no api or 4 if available, and check flash HLS api to see if there’s any equivalent so we can return anything that is actually meaningful.

File: src/flash.js [hunk 3]
```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+
+  /**
+   * @returns {Number} - The height of the video player, in pixels.
+   * @public
+   */
+  get videoHeight(): number {
+    return -1;
```
Comment (OrenMe) [inline src/flash.js]: Check flashhls api to see if we can return true data

File: src/flash.js [hunk 4]
```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+  }
+
+  /**
+   * @returns {Number} - The width of the video player, in pixels.
+   * @public
+   */
+  get videoWidth(): number {
+    return -1;
```
Comment (OrenMe) [inline src/flash.js]: Same comment as for height

File: src/flash.js [hunk 5]
```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+  /**
+   * @param {boolean} playsinline - Whether to set on the video tag the playsinline attribute.
+   */
+  set playsinline(playsinline: boolean): void {}
+
+  /**
+   * @returns {boolean} - Whether the video tag has an attribute of playsinline.
+   */
+  get playsinline(): boolean {
+    return this._config.playsinline;
```
Comment (OrenMe) [inline src/flash.js]: It is meaningless for flash but always return true here

File: src/flash.js [hunk 6]
```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+  }
+
+  /**
+   * Set crossOrigin attribute.
+   * @param {?string} crossOrigin - 'anonymous' or 'use-credentials'
+   */
+  set crossOrigin(crossOrigin: ?string): void {}
```
Comment (OrenMe) [inline src/flash.js]: If something returns then always have return, even undefined.
Future eslint version that we will upgrade to will fail in this

File: src/flash.js [hunk 7]
```diff
@@ -162,11 +169,16 @@ class Flash extends FakeEventTarget implements IEngine {
 
   _init(source: PKMediaSourceObject, config: Object): void {
     this._eventManager = new EventManager();
+    this._config = config;
+    this._loadPromise = null;
```
Comment (OrenMe) [inline src/flash.js]: Define as class member

File: src/flash.js [hunk 8]
```diff
@@ -162,11 +169,16 @@ class Flash extends FakeEventTarget implements IEngine {
...
     if (this._el) {
       this._api = new FlashHLSAdapter(source, config, this._el);
       this._api.attach();
       this._addBindings();
       this._srcToLoad = source.url;
+    } else {
+      this._api = null;
```
Comment (OrenMe) [inline src/flash.js]: Define as class member

File: src/flash.js [hunk 9]
```diff
@@ -162,11 +169,16 @@ class Flash extends FakeEventTarget implements IEngine {
...
+      this._srcToLoad = null;
```
Comment (OrenMe) [inline src/flash.js]: Define as class member

File: src/flash.js (hunk 1 above)
Comment (Yuvalke) [reply src/flash.js]: You're right I saw it now it came from youtube engine and it defined like that.

File: src/flash.js (hunk 5 above)
Comment (Yuvalke) [reply src/flash.js]: Why? I take the value from config as we have in youtube

File: src/flash.js (hunk 6 above)
Comment (OrenMe) [reply src/flash.js]: @Yuvalke ?

File: src/flash.js [hunk 10]
```diff
@@ -621,6 +660,162 @@ class Flash extends FakeEventTarget implements IEngine {
...
+    }
+    return 4;
+  }
+
+  /**
+   * @returns {Number} - The height of the video player, in pixels. flash object will take 100 of this element
+   * @public
+   */
+  get videoHeight(): number {
+    return this._el.getBoundingClientRect().height;
```
Comment (OrenMe) [inline src/flash.js]: Is there no flash HLS api to get this info?

File: src/flash.js (hunk 6 above)
Comment (Yuvalke) [reply src/flash.js]: void

File: src/flash.js (hunk 10 above)
Comment (Yuvalke) [reply src/flash.js]: No, it takes 100% from this element

Comment (OrenMe) [reply src/flash.js]: Please read what it means in HTML5 api https://developer.mozilla.org/en-US/docs/Web/API/HTMLVideoElement/videoWidth
I meant to check if we can get this real Intrinsic width and height and not the doc width and height.
If we can’t get it then return like youtube.

**Pull request merged**